### PR TITLE
Deprecating dynamic_axes and always setting it

### PR DESCRIPTION
**Pull request by corey-nm**
Reasoning: in torch 1.12 any shape operations *without* batch dimension as dynamic will be treated as a constant, and therefore constant folding operations will apply to them. For example in resnet there is a flatten operation towards the end of the graph.

In torch 1.12, without dynamic_axes="batch" set, you get this graph:

image

With dynamic_axes="batch set, you get this graph:

image

### CODE REVIEW

**Review by KSGulin — APPROVED** [review]
LGTM!

File: src/sparseml/pytorch/utils/exporter.py [hunk 1]
```diff
@@ -484,13 +484,18 @@ def export_onnx(
...
-    if dynamic_axes == "batch":
-        dynamic_axes = {
-            tensor_name: {0: "batch"}
-            for tensor_name in (
-                export_kwargs["input_names"] + export_kwargs["output_names"]
-            )
-        }
+    if dynamic_axes is not None:
+        warnings.warn(
+            "Setting dynamic_axes is deprecated, now it is always set",
```
Comment (bfineran) [inline src/sparseml/pytorch/utils/exporter.py:489]: let's make this warning a bit more specific to show that the value the user set `dynamic_axes` as is no longer respected and we are setting the 0th index across all inputs as dynamic

**Review by bfineran — CHANGES_REQUESTED** [review]

File: src/sparseml/pytorch/utils/exporter.py (hunk 1 above)
Comment (corey-nm) [reply src/sparseml/pytorch/utils/exporter.py]: Updated

**Review by rahul-tuli — APPROVED** [review]

**Review by bfineran — APPROVED** [review]

**Pull request merged**